korge/src/commonMain/kotlin/com/soywiz/korge3d/BatchBuilder3D.kt

```diff
@@ -0,0 +1,497 @@
+package com.soywiz.korge3d
+
+import com.soywiz.klogger.*
+import com.soywiz.kmem.*
+import com.soywiz.korag.*
+import com.soywiz.korag.shader.*
+import com.soywiz.korge.internal.*
+import com.soywiz.korge.render.*
+import com.soywiz.korge.view.*
+import com.soywiz.korim.bitmap.*
+import com.soywiz.korim.color.*
+import com.soywiz.korio.async.*
+import com.soywiz.korma.geom.*
+import kotlin.math.*
+
+private val logger = Logger("BatchBuilder3D")
+
+/**
+ * Allows to draw quads and sprites buffering the geometry to limit the draw batches executed calling [AG] (Accelerated Graphics).
+ * This class handles a vertex structure of: x, y, u, v, colorMul, colorAdd. Allowing to draw texturized and tinted primitives.
+ *
+ * You should call: [drawQuad], [drawQuadFast], [drawNinePatch], [drawVertices] for buffering the geometries.
+ *
+ * For performance the actual drawing/rendering doesn't happen until the [flush] method is called (normally that happens automatically).
+ * Then the engine will call [flush] when required, automatically once the buffer is filled, at the end of the frame, or when [RenderContext.flush] is executed
+ * by other renderers.
+ */
+class BatchBuilder3D(
```

Comment:
Why is this class required? Seems a copy of BatchBuilder2D. Since I cannot get the diff, which are the differences, and why is this needed?

Reply:
I think it is no longer needed.
I was using it as part of the 3D Text impl.
But have had to change the approach.
I did say that the 3DText is still WIP. (maybe it should be on a separate PR)

Reply:
I see. If this is not required anymore, maybe this can be removed.

Or if you need something very similar to BatchBuilder2D, we can rename BatchBuilder2D to just BatchBuilder, then make that open or abstract so we can reuse the code. Alternatively if we just need some extension methods, maybe the BatchBuilder could be closed, which usually gives better performance on native.